unumpy/_multimethods.py

```diff
@@ -1132,6 +1256,25 @@ def column_stack(tup):
     return tup
 
 
+def _dstack_default(tup):
+    arrays = []
+    for arr in tup:
+        if ndim(arr) == 1:
+            arrays.append(expand_dims(arr, (0, -1)))
+        elif ndim(arr) == 2:
```

Comment:
You should call `ndim` only once and store the result:
```suggestion
        nd = ndim(arr)
        if nd == 1:
            arrays.append(expand_dims(arr, (0, -1)))
        elif nd == 2:
```